src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Formatting/Rules/TokenBasedFormattingRule.cs

```diff
@@ -86,6 +86,9 @@ public override AbstractFormattingRule WithOptions(SyntaxFormattingOptions optio
 
                     if (!previousToken.IsCloseBraceOfExpression())
                     {
+                        if (currentToken.IsKind(SyntaxKind.SemicolonToken) && currentToken.Parent is EmptyStatementSyntax)
+                            return CreateAdjustNewLinesOperation(1, AdjustNewLinesOption.PreserveLines);
```

Comment:
❓ Is this formatting rule always active, or only active when applying the fix for this diagnostic?

Reply:
this is only active if the token has elastic trivia on it.  So it would be active for this particular fix, or any fix that either created a fres-semicolon token in this fashion (or explicitly added elastic trivia to an existing semicolon).  

this is the behavior we want.  any generated code should place the semicolon for an empty-statement on its own line.